src/main/kotlin/io/github/nomisrev/validation.kt

```diff
@@ -149,3 +150,13 @@ private val emailPattern = ".+@.+\\..+".toRegex()
 
 private fun String.looksLikeEmail(): EitherNel<String, String> =
   if (emailPattern.matches(this)) right() else "'$this' is invalid email".leftNel()
+
+fun NewArticle.validate(): Either<IncorrectInput, NewArticle> =
+  zipOrAccumulate(
+      title.validTitle(),
+      description.validDescription(),
+      body.validBody(),
+      validTags(tagList ?: listOf()).map { it.toList() },
```

Comment:
hi @nomisRev , is this the correct method for dealing with the nullish nature of 'tagList'?

Reply:
Yes, that would indeed be the correct way of dealign with the nullable nature. There is also an extension method `orEmpty()` but TBH I prefer `?: listOf` or `?: emptyList()` in this case.
Side-note: `listOf()` is optimised to `emptyList()` singleton.

but due to the change of `= emptyList()` `tagList` is no longer nullable.

```suggestion
validTags(tagList).map { it.toList() },
```

I looked into the existing code, and I would update `validTags` to return ` EitherNel<InvalidTag, List<String>>` instead of ` EitherNel<InvalidTag, Set<String>>`. That way there is less redundant operations, and we can replace the code to:

```suggestion
validTags(tagList),
```